lib/improver/utilities/cube_manipulation.py

```diff
@@ -524,18 +524,48 @@ def _equalise_cell_methods(cubes):
     return cubelist
 
 
-def compare_attributes(cubes):
+def get_filtered_attributes(cube, attribute_filter=None):
+    """
+    Build dictionary of attributes that match the attribute_filter. If the
+    attribute_filter is None, return all attributes.
+
+    Args:
+        cube (iris.cube.Cube):
+            A cube from which attributes partially matching the
+            self.grid_metadata_identifier will be returned.
```

Comment:
I don't think `self.grid_metadata_identifier` is relevant now that this function is quite general. Should this be `attribute_filter`?